### PR TITLE
Ported to linter-plus.

### PR DESCRIPTION
**Pull request by keplersj**
Merge when linter-plus is published.

### CODE REVIEW

File: lib/linter-swiftlint.coffee [hunk 1]
```diff
@@ -1,40 +1,54 @@
...
 
-  constructor: (editor) ->
-    super(editor)
-    @listen = atom.config.observe 'linter-swiftlint.liveLinting', (value) =>
-      @lintLive = value
+  provideLinter: ->
+    {
+      scopes: ['source.swift']
+      lint: @lint
+      lintOnFly: true
```
Comment (steelbrain) [inline lib/linter-swiftlint.coffee]: shouldn't this be false? 'cause you are giving the file to the program and `lintOnFly` means that your validator accepts the current text buffer as stdin and lints it. Which doens't seem to be the case here.

Comment (keplersj) [inline lib/linter-swiftlint.coffee]: Damn. You caught my 3 AM brain. I'll fix that when I get to my machine later today.

**Comment by keplersj** [comment]
Merge after AtomLinter/linter-plus#24

**Comment by steelbrain** [comment]
Bump.

**Comment by keplersj** [comment]
I need to up the API version number, then I'll publish.

File: lib/linter-swiftlint.coffee [hunk 2]
```diff
@@ -1,40 +1,53 @@
...
+    XRegExp = require('xregexp').XRegExp
 
-  destroy: ->
-    @listen.dispose()
+    regex = XRegExp('(?<file>.+):(?<line>\\d+):\\s(?<type>\\w+):\\s(?<message>.*)')
 
-module.exports = LinterSwiftlint
+    return new Promise (Resolve) ->
+      FilePath = TextEditor.getPath()
+      return unless FilePath # Files that have not be saved
```
Comment (steelbrain) [inline lib/linter-swiftlint.coffee:27]: This line is redundant

**Comment by steelbrain** [comment]
Bump. This is super-old.

**Comment by keplersj** [comment]
I'll finish this I promise.

**Comment by keplersj** [comment]
It's on my list of things to do. It's just not very incentivized since not many people use this.

**Pull request closed without merging**